src/docs/sphinx/dev_guide/gpu_porting.rst

```diff
@@ -0,0 +1,302 @@
+.. ## Copyright (c) 2017-2022, Lawrence Livermore National Security,con LLC and
+.. ## other Axom Project Developers. See the top-level LICENSE file for details.
+.. ##
+.. ## SPDX-License-Identifier: (BSD-3-Clause)
+
+.. _gpu-porting-label:
+
+*******************************
+GPU Porting in Axom
+*******************************
+
+Axom uses RAJA and Umpire as the main workhorses for GPU porting,
+with a set of convenience macros and `axom`-namespaced wrappers encapsulating
+commonly used RAJA and Umpire functions, and preset execution spaces for
+host/device execution.
+
+For the user's guide on using GPU utilities, see also
+:ref:`Core Acceleration<core-acceleration>`
+
+
+.. _gpu-macros-label:
+
+===============
+Macros 
+===============
+
+Axom's macros can be found in file
+`axom/core/Macros.hpp <https://github.com/LLNL/axom/blob/develop/src/axom/core/Macros.hpp>`_
+
+Most of the GPU-related macros are used to guard device code for compilation
+time or for ``__host__ __device__`` decoration of functions/lambdas.
+
+For guarding device code:
+
+.. literalinclude:: ../../../axom/core/Macros.hpp
+   :start-after:  _guarding_macros_start
+   :end-before:  _guarding_macros_end
+   :language: C++
+
+For ``__host__`` or `` __device__`` decoration, or both:
+
+.. literalinclude:: ../../../axom/core/Macros.hpp
+   :start-after:  _decorating_macros_start
+   :end-before:  _decorating_macros_end
+   :language: C++
+
+.. _gpu-memory-label:
+
+===============
+Memory 
+===============
+
+Axom's memory management routines can be found in the file
+`axom/core/memory_management.hpp <https://github.com/LLNL/axom/blob/develop/src/axom/core/memory_management.hpp>`_
+
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Memory Management Routines
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Umpire has the concept of "allocators" associated with each
+`memory resource type`_ (``umpire::resource::MemoryResourceType``)
+
+To allocate memory on a particular resource, you use the ID for the allocator
+associated with the MemoryResourceType.
+
+You are able to set a default allocator, whereby all your memory allocations
+will go on the resource associated with the allocator unless otherwise
+specified::
+
+  //---------------------------------------------------------------------------
```

Comment:
This is code highlighting as Python I think. Please change it so it is specifically C++.